android-components/components/browser/state/src/main/java/mozilla/components/browser/state/engine/middleware/TranslationsMiddleware.kt

```diff
@@ -466,6 +471,47 @@ class TranslationsMiddleware(
         )
     }
 
+    /**
+     * Fetches the expected translation model download size assuming the user intends to complete
+     * a translation using detected defaults.
+     *
+     * If detected defaults are available, then this will sets the values on
+     * [SessionState.translationsState] on the [BrowserStore]
+     * via [TranslationsAction.FetchTranslationDownloadSizeAction].
+     *
+     * If no defaults are available, then no action will occur.
+     *
+     * @param context Context to use to dispatch to the store.
+     * @param tabId Tab ID associated with the request.
+     *
+     */
+    private fun requestDefaultModelDownloadSize(
+        context: MiddlewareContext<BrowserState, BrowserAction>,
+        tabId: String,
+    ) {
+        val detectedLanguages = context.store.state.findTab(tabId)
+            ?.translationsState?.translationEngineState?.detectedLanguages
+        val supportedLanguages = context.store.state.translationEngine.supportedLanguages
+        val fromLanguage = detectedLanguages?.documentLangTag?.let { fromTag ->
+            supportedLanguages?.findLanguage(fromTag)
+        }
```

Comment:
```suggestion
        val fromLanguage = detectedLanguages?.documentLangTag?.let { fromTag ->
            supportedLanguages?.findLanguage(fromTag)
        } ?: return
```
If we hit null at any point, we might as well just escape instead of continuing to do more work.